src/core/cdr/src/dds_cdrstream.c

```diff
@@ -954,6 +955,7 @@ static uint32_t read_union_discriminant (dds_istream_t * __restrict is, uint32_t
       break;
     default: return 0;
```

Comment:
I would suggest the following which won't actually change anything in how the function works now.
```suggestion
    default: break;
   }
   return 0;
```